src/library_egl.js

```diff
@@ -27,16 +49,69 @@ var LibraryEGL = {
         EGL.setErrorCode(0x3008 /* EGL_BAD_DISPLAY */);
         return 0;
       }
-      // TODO: read attribList.
       if ((!config || !config_size) && !numConfigs) {
         EGL.setErrorCode(0x300C /* EGL_BAD_PARAMETER */);
         return 0;
       }
-      if (numConfigs) {
-        {{{ makeSetValue('numConfigs', '0', '1', 'i32') }}}; // Total number of supported configs: 1.
+
+      // config variables to compare against
+      var alpha = 0;
+      var samples = 0;
+      var buffers = 0;
+      var depth = 0;
+      var stencil = 0;
+
+      if (attribList) {
+        while (1) {
+          var param = {{{ makeGetValue('attribList', '0', 'i32') }}};
+          if (param == 0x3038) { // EGL_NONE
```

Comment:
I added a sanity check for the attribute ID parameter. This should capture the vast majority of bad eglChooseConfig calls; I toyed with checking the values as well, but the EGL spec is extremely vague when it comes to validation requirements.